aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -266,7 +266,10 @@ Tensor& rsqrt_out(Tensor& result, const Tensor& self) { return unary_op_impl_out
 Tensor rsqrt(const Tensor& self) { return unary_op_impl(self, at::rsqrt_out); }
 Tensor& rsqrt_(Tensor& self) { return unary_op_impl_(self, at::rsqrt_out); }
 
-Tensor& sign_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, sign_stub); }
+Tensor& sign_out(Tensor& result, const Tensor& self) {
+  TORCH_CHECK(!self.is_complex(), "Unlike NumPy, torch.sign is not intended to support complex numbers.");
```

Comment:
I don't think we need to add this. The current error is sufficient, however it's good to have a test to check that sign throws an error for complex.
```
>>> a=torch.randn(4, dtype=torch.cdouble)
>>> a.sign()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "sign_cpu" not implemented for 'ComplexDouble'
```

Reply:
Perhaps you can change the message in your #39955 ? This at least points users to the proper function.

Reply:
sorry I don't follow. which message are you suggesting to update?

Reply:
This message that I added here: `"Unlike NumPy, torch.sign is not intended to support complex numbers. Please use sgn instead."`

Reply:
I don't think we need the warning message. cc. @mruberry

Reply:
Redirecting users to look at torch.sgn (when it becomes available) seems like a helpful error message. It could be separated from the torch.sgn PR (as a follow-up like this). I would wait to add an error message here until torch.sgn is available, though.

Reply:
I agree that we should either wait until #39955 is merged, or having this message included in #39955 .

Reply:
@xuhdev can you update the error message to what you mentioned above:
`"Unlike NumPy, torch.sign is not intended to support complex tensors. Please use torch.sgn() instead."`